CsCore/xUnitTests/src/Plugins/CsCoreXUnitTests/com/csutil/tests/http/apis/OpenAiTests.cs

```diff
@@ -207,6 +209,98 @@ public class OpenAiTests {
             Log.d("messages=" + JsonWriter.AsPrettyString(messages));
         }
 
+        [Fact]
+        public async Task ExampleUsage6_analyseImage() {
+
+            var openAi = new OpenAi(await IoC.inject.GetAppSecrets().GetSecret("OpenAiKey"));
+            int iterationThreshold = 5;
+            List<string> imageUrls = new List<string>();
+            for (int iteration = 0; iteration < iterationThreshold; iteration++) {
+                var prompt = "A fascinating image from a children's storybook";
+
+                var url = await GenerateFascinatingImage(openAi, prompt);
+                imageUrls.Add(url);
+                Assert.NotEmpty(url);
+
+                List<string> questions = await GenerateQuestionsBasedOnPrompt(openAi, prompt);
+
+                // Check that Ai gave back at least one question
+                Assert.NotEmpty(questions);
+
+
+                var confidenceResponseFormat = new ConfidenceResponse() {
+                    responseConfidences = new Dictionary<string, int>{
+                    {"Does the image evoke a sense of wonder and imagination suitable for a children's storybook?", 100},
+                    {"Are the colors vibrant and appealing to a younger audience?", 70},
+                    {"Does the input image show a picture of a children's book?", 100},
+                    {"Is there a dog in the picture?", 5}
+                }
+                };
+
+                var messages = new List<VisionGpt.Line>() {
+                new VisionGpt.Line(VisionGpt.Role.system, content: "You are a helpful assistant designed to output JSON.")
+            };
+
+                messages.AddQuestionsToImage(url, questions);
+                messages.AddUserLineWithJsonResultStructure("Rate the following questions with a confidence from 0 to 100 based on how well the question fits the image", confidenceResponseFormat);
+
+                // Send the messages to the AI and get the response:
+                var response = await openAi.ImageToText(new VisionGpt.Request(messages));
+                VisionGpt.Line newLine = response.choices.Single().message;
+                messages.Add(newLine);
+
+                // Parse newLine.content as a YesNoResponse:
+                var confidencesResponse = newLine.ParseNewLineContentAsJson<ConfidenceResponse>();
+
+                var isConfident = IsCofidenceHighEnough(confidencesResponse.responseConfidences.Values.ToList());
+                if (isConfident) {
+                    break;
+                }
+
+            }
+            string result = JsonConvert.SerializeObject(imageUrls);
+            File.WriteAllText(@".\VisionImages.json", result);
+        }
+        private static async Task<string> GenerateFascinatingImage(OpenAi openAi, string prompt) {
```

Comment:
rename GenerateFascinatingImage to GenerateImage

Reply:
74270f4d65aa64232aa57b4ea01f8fafc6582ee6